src/validator.js

```diff
@@ -29,47 +41,83 @@ export default class Validator {
   }
 
   validateField(field, value, values = {}) {
-    let result = null;
+    let error = null;
     if (this.config.hasOwnProperty(field)) {
-      const fieldRules = keys(this.config[field]);
-      for (let i = 0; i < fieldRules.length; i++) {
-        const rule = fieldRules[i];
-        const options = { values };
-        if (isBoolean(this.config[field][rule])) {
-          options.if = constant(this.config[field][rule]);
-        }
+      error = this.validateRule(
+        'required',
+        field,
+        value,
+        { values, required: true }
+      );
+      const empty = !!error;
+
+      if (error !== null && !this.evaluateIf(field, 'required', values)) {
+        error = null;
+      }
 
-        result = this.validateRule(
-          rule,
-          field,
-          value,
-          assign({}, this.config[field][rule], options)
-        );
-
-        if (result) {
-          result = { field, rule: result, value, config: this.config[field] };
-          this.errors = assign({}, this.errors, { [field]: result });
-          break;
-        } else {
-          if (this.errors.hasOwnProperty(field)) {
-            this.errors = omit(this.errors, field);
+      if (!empty) {
+        const fieldRules = reject(keys(this.config[field]), 'required');
+        for (let i = 0; i < fieldRules.length; i++) {
+          const ruleName = fieldRules[i];
+          error = this.validateRule(
+            ruleName,
+            field,
+            value,
+            assign({}, this.config[field][ruleName], { values })
+          );
+
+          if (error !== null) {
+            break;
           }
         }
       }
     }
-    return result;
+    return error;
   }
 
   validateRule(ruleName, field, value, options) {
-    if (isFunction(get(options, 'if')) && !get(options, 'if')(get(options, 'values', {}))) {
+    let condition = get(options, 'if');
+    if (this.config.hasOwnProperty(field) && isBoolean(this.config[field][ruleName])) {
```

Comment:
This condition is used here and in `evaluateIf`. Could you make it a method with `field` and `ruleName` as parameters and give it a good name to improve readbility?